File: examples/setheading.rs
use slowpoke::*;

fn main() {
    TurtleArgs::default()
        .with_size(400, 400)
        .with_title("heading demo")
        .run(|turtle| {
            turtle.speed(1);
            turtle.dot().with_size(3).with_color("red");
            turtle.teleport(10, 20);
            println!("teleported to 10,20; pos = {:?}", turtle.pos());

            // draw to the right
            turtle.setheading(0);
            turtle.forward(100);
            println!("heading 0; forward 100; pos = {:?}", turtle.pos());

            turtle.teleport(10, 20);
            println!("teleported to 10,20; pos = {:?}", turtle.pos());

            // draw "south" or down
            turtle.setheading(270);
            turtle.forward(100);
            println!("heading 270; forward 100; pos = {:?}", turtle.pos());

            turtle.teleport(10, 20);
            println!("teleported to 10,20; pos = {:?}", turtle.pos());

            // draw "north" or up
            turtle.setheading(90);
            turtle.forward(100);
            println!("pos = {:?}", turtle.pos());
            turtle.teleport(10, 20);

            // draw "west" or to the left
            turtle.setheading(180);
            turtle.forward(100);
            println!("pos = {:?}", turtle.pos());
            turtle.teleport(10, 20);

            turtle.left(30);
        });
}

File: python-examples/setheading.py
import turtle
from time import sleep

turtle.speed(1)
turtle.dot(3, "red")
sleep(10)

turtle.penup()
turtle.goto(10, 20)
turtle.pendown()

# draw to the right
turtle.setheading(0)
turtle.forward(100)
print(f"pos = {turtle.pos()}")

turtle.penup()
turtle.goto(10, 20)
turtle.pendown()

# draw "south" or down
turtle.setheading(270)
turtle.forward(100)
print(f"pos = {turtle.pos}")

turtle.penup()
turtle.goto(10, 20)
turtle.pendown()

# draw "north" or up
turtle.setheading(90)
turtle.forward(100)
print(f"pos = {turtle.pos}")

turtle.penup()
turtle.goto(10, 20)
turtle.pendown()

# draw "west" or to the left
turtle.setheading(180)
turtle.forward(100)
print(f"pos = {turtle.pos}")

turtle.penup()
turtle.goto(10, 20)
turtle.pendown()

turtle.left(30)

turtle.mainloop()
